### PR TITLE
Fixes #12531 proposals

### LINKED ISSUE #12531: last covs
```

Please find the latest report on new defect(s) introduced to radare2 found with Coverity Scan.

9 new defect(s) introduced to radare2 found with Coverity Scan.
23 defect(s), reported by Coverity Scan earlier, were marked fixed in the recent build analyzed by Coverity Scan.

New defect(s) Reported-by: Coverity Scan
Showing 9 of 9 defect(s)


** CID 1397061:  Control flow issues  (NO_EFFECT)
/libr/anal/esil.c: 1214 in esil_asr()


________________________________________________________________________________________________________
*** CID 1397061:  Control flow issues  (NO_EFFECT)
/libr/anal/esil.c: 1214 in esil_asr()
1208     	int regsize = 0, ret = 0;
1209     	ut64 op_num = 0, param_num = 0;
1210     	char *op = r_anal_esil_pop (esil);
1211     	char *param = r_anal_esil_pop (esil);
1212     	if (op && r_anal_esil_get_parm_size (esil, op, &op_num, &regsize)) {
1213     		if (param && r_anal_esil_get_parm (esil, param, &param_num)) {
   CID 1397061:  Control flow issues  (NO_EFFECT)
   This less-than-zero comparison of an unsigned value is never true. "param_num < 0ULL".
1214     			if (param_num < 0 || param_num > regsize - 1) {
1215     				// capstone bug?
1216     				eprintf ("Invalid asr shift of %"PFMT64d" at 0x%"PFMT64x"\n", param_num, esil->address);
1217     				param_num = 30;
1218     			}
1219     			bool isNegative;

** CID 1397060:  Resource leaks  (RESOURCE_LEAK)
/libr/io/p/io_default.c: 279 in r_io_def_mmap_write()


________________________________________________________________________________________________________
*** CID 1397060:  Resource leaks  (RESOURCE_LEAK)
/libr/io/p/io_default.c: 279 in r_io_def_mmap_write()
273     				}
274     				memcpy (a_buf+a_delta, buf, count);
275     				for (i = 0; i < a_count; i += aligned) {
276     					(void)lseek (mmo->fd, a_off + i, SEEK_SET);
277     					len = write (mmo->fd, a_buf + i, aligned);
278     					if (len != aligned) {
   CID 1397060:  Resource leaks  (RESOURCE_LEAK)
   Variable "a_buf" going out of scope leaks the storage it points to.
279     						return len;
280     					}
281     				}
282     			}
283     			free (a_buf);
284     			return count;

** CID 1397059:    (OVERRUN)
/libr/util/str.c: 1532 in r_str_ansi_filter()
/libr/util/str.c: 1532 in r_str_ansi_filter()


________________________________________________________________________________________________________
*** CID 1397059:    (OVERRUN)
/libr/util/str.c: 1532 in r_str_ansi_filter()
1526     		if (tmp[i] == 0x1b) {
1527     			if ((i + 1) < len && tmp[i + 1] == '[') {
1528     				for (i += 2; i < len && str[i] != 'J' && str[i] != 'm' && str[i] != 'H'; i++) {
1529     				;
1530     				}
1531     			}
   CID 1397059:    (OVERRUN)
   Overrunning dynamic array "tmp" at offset corresponding to index variable "i".
1532     			if (tmp[i + 1] == '#' && isdigit (tmp[i + 2])  && tmp[i + 3]) {
1533     				i += 3;
1534     			}
1535     		} else {
1536     			str[j] = tmp[i];
1537     			cps[j] = i;
/libr/util/str.c: 1532 in r_str_ansi_filter()
1526     		if (tmp[i] == 0x1b) {
1527     			if ((i + 1) < len && tmp[i + 1] == '[') {
1528     				for (i += 2; i < len && str[i] != 'J' && str[i] != 'm' && str[i] != 'H'; i++) {
1529     				;
1530     				}
1531     			}
   CID 1397059:    (OVERRUN)
   Overrunning dynamic array "tmp" at offset corresponding to index variable "i".
1532     			if (tmp[i + 1] == '#' && isdigit (tmp[i + 2])  && tmp[i + 3]) {
1533     				i += 3;
1534     			}
1535     		} else {
1536     			str[j] = tmp[i];
1537     			cps[j] = i;

** CID 1397058:  Control flow issues  (NO_EFFECT)
/libr/anal/esil.c: 1178 in esil_asreq()


________________________________________________________________________________________________________
*** CID 1397058:  Control flow issues  (NO_EFFECT)
/libr/anal/esil.c: 1178 in esil_asreq()
1172     					ut64 left_bits = 0;
1173     					int shift = regsize - 1;
1174     					if (shift < 0 || shift > regsize - 1) {
1175     						eprintf ("Invalid asreq shift of %d at 0x%"PFMT64x"\n", shift, esil->address);
1176     						shift = 0;
1177     					}
   CID 1397058:  Control flow issues  (NO_EFFECT)
   This less-than-zero comparison of an unsigned value is never true. "param_num < 0ULL".
1178     					if (param_num < 0 || param_num > regsize - 1) {
1179     						// capstone bug?
1180     						eprintf ("Invalid asreq shift of %"PFMT64d" at 0x%"PFMT64x"\n", param_num, esil->address);
1181     						param_num = 30;
1182     					}
1183     					if (op_num & (1LL << shift)) {

** CID 1397057:  Resource leaks  (RESOURCE_LEAK)
/libr/bin/obj.c: 184 in r_bin_object_new()


________________________________________________________________________________________________________
*** CID 1397057:  Resource leaks  (RESOURCE_LEAK)
/libr/bin/obj.c: 184 in r_bin_object_new()
178     		if (sz < bsz) {
179     			bsz = sz;
180     		}
181     		ut8 *bytes = malloc (sz);
182     		if (!bytes) {
183     			eprintf ("Cannot allocate %" PFMT64u " bytes\n", sz);
   CID 1397057:  Resource leaks  (RESOURCE_LEAK)
   Variable "o" going out of scope leaks the storage it points to.
184     			return NULL;
185     		}
186     		r_buf_read_at (binfile->buf, offset, bytes, sz);
187     		if (!plugin->load_bytes (binfile, &o->bin_obj, bytes, sz,
188     					 loadaddr, sdb)) {
189     			bprintf ("Error in r_bin_object_new: load_bytes failed for %s plugin\n", plugin->name);

** CID 1397056:    (RESOURCE_LEAK)
/libr/core/cbin.c: 1894 in bin_symbols()
/libr/core/cbin.c: 1897 in bin_symbols()
/libr/core/cbin.c: 1900 in bin_symbols()
/libr/core/cbin.c: 1904 in bin_symbols()
/libr/core/cbin.c: 1900 in bin_symbols()
/libr/core/cbin.c: 1904 in bin_symbols()


________________________________________________________________________________________________________
*** CID 1397056:    (RESOURCE_LEAK)
/libr/core/cbin.c: 1894 in bin_symbols()
1888     		char *r_symbol_name = r_str_escape_utf8_for_json (symbol->name, -1);
1889     		ut64 addr = symbol->paddr == UT64_MAX ? symbol->vaddr : rva (r->bin, symbol->paddr, symbol->vaddr, va);
1890     		int len = symbol->size ? symbol->size : 32;
1891     		SymName sn = {0};
1892     
1893     		if (exponly && !isAnExport (symbol)) {
   CID 1397056:    (RESOURCE_LEAK)
   Variable "r_symbol_name" going out of scope leaks the storage it points to.
1894     			continue;
1895     		}
1896     		if (name && strcmp (r_symbol_name, name)) {
1897     			continue;
1898     		}
1899     		if (at && (!symbol->size || !is_in_range (at, addr, symbol->size))) {
/libr/core/cbin.c: 1897 in bin_symbols()
1891     		SymName sn = {0};
1892     
1893     		if (exponly && !isAnExport (symbol)) {
1894     			continue;
1895     		}
1896     		if (name && strcmp (r_symbol_name, name)) {
   CID 1397056:    (RESOURCE_LEAK)
   Variable "r_symbol_name" going out of scope leaks the storage it points to.
1897     			continue;
1898     		}
1899     		if (at && (!symbol->size || !is_in_range (at, addr, symbol->size))) {
1900     			continue;
1901     		}
1902     		if ((printHere && !is_in_range (r->offset, symbol->paddr, len))
/libr/core/cbin.c: 1900 in bin_symbols()
1894     			continue;
1895     		}
1896     		if (name && strcmp (r_symbol_name, name)) {
1897     			continue;
1898     		}
1899     		if (at && (!symbol->size || !is_in_range (at, addr, symbol->size))) {
   CID 1397056:    (RESOURCE_LEAK)
   Variable "r_symbol_name" going out of scope leaks the storage it points to.
1900     			continue;
1901     		}
1902     		if ((printHere && !is_in_range (r->offset, symbol->paddr, len))
1903     				&& (printHere && !is_in_range (r->offset, addr, len))) {
1904     			continue;
1905     		}
/libr/core/cbin.c: 1904 in bin_symbols()
1898     		}
1899     		if (at && (!symbol->size || !is_in_range (at, addr, symbol->size))) {
1900     			continue;
1901     		}
1902     		if ((printHere && !is_in_range (r->offset, symbol->paddr, len))
1903     				&& (printHere && !is_in_range (r->offset, addr, len))) {
   CID 1397056:    (RESOURCE_LEAK)
   Variable "r_symbol_name" going out of scope leaks the storage it points to.
1904     			continue;
1905     		}
1906     		snInit (r, &sn, symbol, lang);
1907     
1908     		if (IS_MODE_SET (mode) && (is_section_symbol (symbol) || is_file_symbol (symbol))) {
1909     			/*
/libr/core/cbin.c: 1900 in bin_symbols()
1894     			continue;
1895     		}
1896     		if (name && strcmp (r_symbol_name, name)) {
1897     			continue;
1898     		}
1899     		if (at && (!symbol->size || !is_in_range (at, addr, symbol->size))) {
   CID 1397056:    (RESOURCE_LEAK)
   Variable "r_symbol_name" going out of scope leaks the storage it points to.
1900     			continue;
1901     		}
1902     		if ((printHere && !is_in_range (r->offset, symbol->paddr, len))
1903     				&& (printHere && !is_in_range (r->offset, addr, len))) {
1904     			continue;
1905     		}
/libr/core/cbin.c: 1904 in bin_symbols()
1898     		}
1899     		if (at && (!symbol->size || !is_in_range (at, addr, symbol->size))) {
1900     			continue;
1901     		}
1902     		if ((printHere && !is_in_range (r->offset, symbol->paddr, len))
1903     				&& (printHere && !is_in_range (r->offset, addr, len))) {
   CID 1397056:    (RESOURCE_LEAK)
   Variable "r_symbol_name" going out of scope leaks the storage it points to.
1904     			continue;
1905     		}
1906     		snInit (r, &sn, symbol, lang);
1907     
1908     		if (IS_MODE_SET (mode) && (is_section_symbol (symbol) || is_file_symbol (symbol))) {
1909     			/*

** CID 1397055:  Resource leaks  (RESOURCE_LEAK)
/libr/bin/p/bin_wasm.c: 229 in symbols()


________________________________________________________________________________________________________
*** CID 1397055:  Resource leaks  (RESOURCE_LEAK)
/libr/bin/p/bin_wasm.c: 229 in symbols()
223     	return ret;
224     bad_alloc:
225     	// not so sure if imports should be freed.
226     	r_list_free (symtab);
227     	r_list_free (codes);
228     	r_list_free (ret);
   CID 1397055:  Resource leaks  (RESOURCE_LEAK)
   Variable "exports" going out of scope leaks the storage it points to.
229     	return NULL;
230     }
231     
232     static RList *imports(RBinFile *bf) {
233     	RBinWasmObj *bin = NULL;
234     	RList *imports = NULL;

** CID 1378277:  Incorrect expression  (SIZEOF_MISMATCH)
/libr/magic/apprentice.c: 954 in parse()


________________________________________________________________________________________________________
*** CID 1378277:  Incorrect expression  (SIZEOF_MISMATCH)
/libr/magic/apprentice.c: 954 in parse()
948     
949     			maxmagic += ALLOC_INCR;
950     			if (!(mp = realloc (*mentryp, sizeof (*mp) * maxmagic))) {
951     				file_oomem (ms, sizeof (*mp) * maxmagic);
952     				return -1;
953     			}
   CID 1378277:  Incorrect expression  (SIZEOF_MISMATCH)
   Passing argument "&mp[*nmentryp]" of type "struct r_magic_entry *" and argument "3200UL" ("16UL * 200UL") to function "memset" is suspicious because "sizeof (struct r_magic_entry) /*16*/" is expected.
954     			(void)memset(&mp[*nmentryp], 0, sizeof (*mp) * ALLOC_INCR);
955     			*mentryp = mp;
956     		}
957     		me = &(*mentryp)[*nmentryp];
958     		if (!me->mp) {
959     			if (!(m = malloc (sizeof (*m) * ALLOC_CHUNK))) {

** CID 1288752:    (RESOURCE_LEAK)
/libr/lang/p/pipe.c: 116 in lang_pipe_run()
/libr/lang/p/pipe.c: 120 in lang_pipe_run()


________________________________________________________________________________________________________
*** CID 1288752:    (RESOURCE_LEAK)
/libr/lang/p/pipe.c: 116 in lang_pipe_run()
110     	int child, ret;
111     	int input[2];
112     	int output[2];
113     
114     	if (pipe (input) != 0) {
115     		eprintf ("r_lang_pipe: pipe failed on input\n");
   CID 1288752:    (RESOURCE_LEAK)
   Handle variable "safe_in" going out of scope leaks the handle.
116     		return false;
117     	}
118     	if (pipe (output) != 0) {
119     		eprintf ("r_lang_pipe: pipe failed on output\n");
120     		return false;
121     	}
/libr/lang/p/pipe.c: 120 in lang_pipe_run()
114     	if (pipe (input) != 0) {
115     		eprintf ("r_lang_pipe: pipe failed on input\n");
116     		return false;
117     	}
118     	if (pipe (output) != 0) {
119     		eprintf ("r_lang_pipe: pipe failed on output\n");
   CID 1288752:    (RESOURCE_LEAK)
   Handle variable "safe_in" going out of scope leaks the handle.
120     		return false;
121     	}
122     	
123     	env ("R2PIPE_IN", input[0]);
124     	env ("R2PIPE_OUT", output[1]);
125     
```

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: libr/anal/esil.c [hunk 1]
```diff
@@ -1175,7 +1175,7 @@ static int esil_asreq(RAnalEsil *esil) {
 						eprintf ("Invalid asreq shift of %d at 0x%"PFMT64x"\n", shift, esil->address);
 						shift = 0;
 					}
```
Comment (radare) [inline libr/anal/esil.c]: Not sure if this is correct

File: libr/anal/esil.c [hunk 2]
```diff
@@ -1211,7 +1211,7 @@ static int esil_asr(RAnalEsil *esil) {
 	char *param = r_anal_esil_pop (esil);
 	if (op && r_anal_esil_get_parm_size (esil, op, &op_num, &regsize)) {
 		if (param && r_anal_esil_get_parm (esil, param, &param_num)) {
-			if (param_num < 0 || param_num > regsize - 1) {
```
Comment (radare) [inline libr/anal/esil.c]: Same here

File: libr/anal/esil.c (hunk 1 above)
Comment (devnexen) [reply libr/anal/esil.c]: If they re unsigned ... makes sense ?

Comment (radare) [reply libr/anal/esil.c]: ok missed that detail

**Comment by radare** [comment]
Btw use better commit messages:

* If you fix an issue name it like "Fix #... - description of the fix"
* Always start in uppercase
* Use ## if the commit is important for the release notes

**Pull request merged**